app/src/main/java/m/co/rh/id/a_news_provider/app/provider/AbstractRSSContentProvider.java

```diff
@@ -0,0 +1,352 @@
+package m.co.rh.id.a_news_provider.app.provider;
+
+import android.content.ContentProvider;
+import android.content.ContentValues;
+import android.content.UriMatcher;
+import android.database.AbstractCursor;
+import android.database.Cursor;
+import android.net.Uri;
+
+import androidx.annotation.NonNull;
+
+import java.util.List;
+
+/**
+ * RSS Feed {@link ContentProvider} will provide read access to RSS feeds and articles.
+ * <p>
+ * URI is like this:
+ * <ul>
+ *     <li>/feeds - will return a list of feeds, each of which is a pair of an id and a name.</li>
+ *     <li>/articles - will return a list of all articles regardless of feed.</li>
+ *     <li>/articles/<feedid> - will return a list of articles, each of which has an id, a title and a text.</li>
+ * </ul>
+ */
+public abstract class AbstractRSSContentProvider extends ContentProvider {
+
+    /**
+     * The column names in the response to the "feeds" URI.
+     */
+    private final String[] FEED_COL_NAMES = {"id", "title"};
+
+    /**
+     * The column names in the response to the "articles/*" URI.
+     */
+    private final String[] ARTICLE_COL_NAMES = {"id", "title", "text"};
```

Comment:
Why?
I like to mention my argument against it: Other implementations already use "text" and having a consistent API for the content provider is nice for clients trying to play with multiple RSS reader apps.

Reply:
What you mean by other implementation? You mean other apps? "text" doesnt make sense here since the data was a list of articles. "body" or "content" would be more precise.